bundles/binding/org.openhab.binding.dscalarm/src/main/java/org/openhab/binding/dscalarm/internal/model/Panel.java

```diff
@@ -32,131 +33,109 @@
 public class Panel extends DSCAlarmDevice {
     private static final Logger logger = LoggerFactory.getLogger(Panel.class);
 
-    public DSCAlarmDeviceProperties panelProperties = new DSCAlarmDeviceProperties();
+    private int panelID;
+    private Date panelTime = new Date(0);
 
     /**
      * Constructor
      *
      * @param panelId
      */
-    public Panel(int panelId) {
-        panelProperties.setPanelId(panelId);
+    public Panel(int panelID) {
+        this.panelID = panelID;
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public void refreshItem(Item item, DSCAlarmBindingConfig config, EventPublisher publisher) {
+    public void refreshItem(Item item, DSCAlarmBindingConfig config, EventPublisher publisher, int state, String description) {
         logger.debug("refreshItem(): Panel Item Name: {}", item.getName());
 
-        int state;
         String str = "";
-        boolean trigger;
-        boolean trouble;
-        boolean boolState;
         OnOffType onOffType;
 
         if (config != null) {
             if (config.getDSCAlarmItemType() != null) {
                 switch (config.getDSCAlarmItemType()) {
                     case PANEL_CONNECTION:
-                        state = panelProperties.getSystemConnection();
                         publisher.postUpdate(item.getName(), new DecimalType(state));
                         break;
                     case PANEL_MESSAGE:
-                        str = panelProperties.getSystemMessage();
-                        publisher.postUpdate(item.getName(), new StringType(str));
+                        publisher.postUpdate(item.getName(), new StringType(description));
                         break;
                     case PANEL_SYSTEM_ERROR:
-                        str = String.format("%03d", panelProperties.getSystemErrorCode()) + ": " + panelProperties.getSystemErrorDescription();
+                        str = String.format("%03d", state) + ": " + ((state == 0) ? "No Error" : description);
```

Comment:
Minor: it could be more efficient to String.format the entire string, not just part and then use string concatenation.